src/rust/ide/enso-protocol/src/language_server/tests.rs

```diff
@@ -0,0 +1,490 @@
+    executor  : futures::executor::LocalPool,
+}
+
+fn setup_language_server() -> Fixture {
+    let transport = MockTransport::new();
+    let client    = Client::new(transport.clone());
+    let executor  = futures::executor::LocalPool::new();
+    executor.spawner().spawn_local(client.runner()).unwrap();
+    Fixture {transport,client,executor}
+}
+
+#[test]
+fn test_file_event_notification() {
+    let mut fixture = setup_language_server();
+    let mut events  = Box::pin(fixture.client.events());
+    assert!(poll_stream_output(&mut events).is_none());
+
+    let root_id = uuid::Uuid::parse_str("00000000-0000-0000-0000-000000000000");
+    let root_id = root_id.expect("Couldn't parse uuid.");
+    let expected_event = FileEvent {
+        path : Path{root_id,segments:vec!["Main.txt".into()]},
+        kind : FileEventKind::Modified,
+    };
+    let notification_text = r#"{
+            "jsonrpc": "2.0",
+            "method": "file/event",
+            "params": {
+                "event" : {
+                    "path" : {
+                        "rootId"   : "00000000-0000-0000-0000-000000000000",
+                        "segments" : ["Main.txt"]
+                    },
+                    "kind" : "Modified"
+                }
+            }
+        }"#;
+    fixture.transport.mock_peer_message_text(notification_text);
+    assert!(poll_stream_output(&mut events).is_none());
+
+    fixture.executor.run_until_stalled();
+
+    let event = poll_stream_output(&mut events);
+    if let Some(Event::Notification(n)) = event {
+        assert_eq!(n, Notification::FileEvent {event:expected_event});
+    } else {
+        panic!("expected notification event");
+    }
+}
+
+/// This function tests making a request using language server. It
+/// * creates FM client and uses `make_request` to make a request,
+/// * checks that request is made for `expected_method`,
+/// * checks that request input is `expected_input`,
+/// * mocks receiving a response from server with `result` and
+/// * checks that FM-returned Future yields `expected_output`.
+fn test_request<Fun, Fut, T>
+( make_request:Fun
+    , expected_method:&str
+    , expected_input:Value
+    , result:Value
+    , expected_output:T )
+    where Fun : FnOnce(&mut Client) -> Fut,
+          Fut : Future<Output = Result<T>>,
+          T   : Debug + PartialEq {
+    let mut fixture        = setup_language_server();
+    let mut request_future = Box::pin(make_request(&mut fixture.client));
+
+    let request = fixture.transport.expect_message::<RequestMessage<Value>>();
+    assert_eq!(request.method, expected_method);
+    assert_eq!(request.params, expected_input);
+
+    let response = Message::new_success(request.id, result);
+    fixture.transport.mock_peer_message(response);
+    fixture.executor.run_until_stalled();
+    let output = poll_future_output(&mut request_future).unwrap().unwrap();
+    assert_eq!(output, expected_output);
+}
+
+#[test]
+fn test_file_requests() {
```

Comment:
This test fn is sooooo long. Can you spliti t into several fans? Im ok if not, but if its possible, pls do it